packages/plugin/src/ChartPlugin.js

```diff
@@ -122,76 +40,41 @@ const ChartPlugin = ({
                 )
             }
         },
-        [canvasRef, visualization, onChartGenerated]
-    )
-
-    const doFetch = useCallback(
-        (visualization, filters, forDashboard) => {
-            fetchData({
-                visualization,
-                filters,
-                d2,
-                forDashboard,
-            })
-                .then(result => {
-                    if (result.responses.length) {
-                        onResponsesReceived(result.responses)
-                    }
-
-                    fetchResult.current = result
-                    renderVisualization(defaultAnimation)
-                    onLoadingComplete()
-                })
-                .catch(error => {
-                    onError(error)
-                })
-        },
-        [
-            d2,
-            onResponsesReceived,
-            onLoadingComplete,
-            onError,
-            renderVisualization,
-            defaultAnimation,
-        ]
+        [canvasRef, visualization, onChartGenerated, responses, extraOptions]
     )
 
     useEffect(() => {
-        doFetch(visualization, filters, forDashboard)
+        renderVisualization(defaultAnimation)
+
         /* eslint-disable-next-line react-hooks/exhaustive-deps */
-    }, [visualization, filters, forDashboard])
+    }, [visualization, responses, extraOptions])
 
     useEffect(() => {
         renderVisualization(0)
-    }, [id, style]) /* eslint-disable-line react-hooks/exhaustive-deps */
+
+        /* eslint-disable-next-line react-hooks/exhaustive-deps */
+    }, [renderCounter, style])
 
     return <div ref={canvasRef} style={style} />
 }
 
 ChartPlugin.defaultProps = {
     visualization: {},
     filters: {},
-    forDashboard: false,
     style: {},
     animation: 200,
-    onError: Function.prototype,
     onChartGenerated: Function.prototype,
-    onLoadingComplete: Function.prototype,
-    onResponsesReceived: Function.prototype,
 }
 
 ChartPlugin.propTypes = {
-    d2: PropTypes.object.isRequired,
+    extraOptions: PropTypes.object.isRequired,
+    // legendSets: PropTypes.arrayOf(PropTypes.object).isRequired,
```

Comment:
Uncomment this. See above.

Reply:
@edoardo is it OK to leave this as a comment until @martinkrulltott uses it?  Otherwise we get ESLint errors saying that the prop is defined but never used

Reply:
Sure!